odoo/http.py

```diff
@@ -1059,6 +1059,14 @@ def flatten(self):
             self.response.append(self.render())
             self.template = None
 
+    def set_cookie(self, *args, **kwargs):
+        cookie_type = kwargs.pop('cookie_type', None)
+        if request.db:
+            if not request.env['ir.http']._is_allowed_cookie(cookie_type, args[0] if args else None):
```

Comment:
Isn't args[0] always set?

Reply:
That's also what I thought but then a test failed on runbot because `auth_totp` controller calls it with a named `key`, therefore `key` ends up in the `kwargs` instead. ... now fixing this, to fetch the value correctly. Thank you for raising this.